### PR TITLE
DataTables: Date Range Filter

### PR DESCRIPTION
**Pull request by ghost**
Adds date range filter.

kodz.ca/projects/wet-boew/demos/tables/custom-filter

### CODE REVIEW

File: src/plugins/tables/tables.js [hunk 1]
```diff
@@ -209,6 +209,26 @@ $document.on( "submit", ".wb-tables-filter", function( event ) {
 				$value += ( $value.length > 0 ) ? "|" : "";
 				$value += $elm.val();
 			}
+		} else if ( $elm.is( "input[type='date']" ) && $elm.val() ) {
+			if ( $elm.is( "input[name='dateFrom']" ) ) {
```
Comment (duboisp) [inline src/plugins/tables/tables.js]: ``` input[name='dateFrom'] ``` that don't see configurable.

It will also need to be documented.

**Review by duboisp — CHANGES_REQUESTED** [review]

File: src/plugins/tables/tables.js (hunk 1 above)
Comment (thekodester) [reply src/plugins/tables/tables.js]: If you have suggestion provide them.

**Comment by thekodester** [comment]
Reconfigured the code so little more configurable and updated documentation.

To use make sure any date element contains `start` or `end` in the name attribute. Examples `name="start_date"` or `name="date_start"`.

**Comment by thekodester** [comment]
I add code to prevent invalid date errors in Safari (IE might be affected from what I read online).

Safari see '2016-07-26 09:07:00' as invalid, but '2016-07-26' as valid.

screen shot 2019-03-01 at 10 11 16 am

**Comment by thekodester** [comment]
This would fix Safari, but I feel like it could be over complicating things:

```
obj = obj.replace(/[0-9]{2}\s[0-9]{2}\:/g, function (e) {return e.replace( /\s/g, "T" );});
objDate = new Date( obj );
```

I we'll close the PR, if anyone has a better idea to work around Safari issues, is free to use the code from this.

**Pull request closed without merging**